src/app/model/ChessEngine.ts

```diff
@@ -68,7 +70,78 @@ export class ChessEngine implements ChessModel {
     return side === Side.WHITE ? Side.BLACK : Side.WHITE;
   }
 
+  private returnWirtualProposedBoardAfterMove(chessboard: Chessboard, squareFrom: Square, squareTo: Square): Chessboard {
+    const wirtualPorposedChessboard = _.cloneDeep(chessboard);
+    const { squaresWithPiece: proposedSquaresWithPieces } = wirtualPorposedChessboard;
+
+    const movedPiece = chessboard.onPositionPiece(squareFrom);
+    const positionFrom = `${squareFrom.column}${squareFrom.row}`;
+    const positionTo = `${squareTo.column}${squareTo.row}`;
+    delete proposedSquaresWithPieces[positionFrom];
+    proposedSquaresWithPieces[positionTo] = movedPiece as Piece;
+
+    return wirtualPorposedChessboard;
+  }
+
+  public static kingPosition(chessboard: Chessboard, kingSide: Side): Square | undefined {
+    const { squaresWithPiece: squaresWithPieces } = chessboard;
```

Comment:
Zobacz, że tutaj wyciagasz squaresWithPieces z chessboard, tylko po to, zeby potem na tym operowac i na koniec zwroci wynik. To jest zlamanie 2 zasad: kohezji i Tell - Don't Ask. Ogólnie po prostu jak juz mamy pieces w ChessBoard i wszystko co jest potrzebne to metode kingPosition mozemy dodac tam :) Wtedy nie wyciagamy na zenwatrz squaresWithPieces jak nie trzeba.

Reply:
Nie rozumiem.
Mógłbyś dać mi przykład?
Nie rozumiem o co chodzi...

Reply:
Te zasady to są takie oficjalne? Trzeba je znać?

Reply:
Jak najbardziej :) Ale to juz tak na "czuja" idzie. Po prostu jak najmniej powinienies wyciagac danych z obiektu, a bazowac na jego zachowaniach - czyli metodach. Jesli cos wyciagasz z obiektu a potem poza nim operujesz tylko na jego danych to jak najbardziej powinno to byc w klasie tego obiektu.

Zobacz sobie tutaj ten przyklad z krokodylem (moze wiecej nie - bez juz kolejnych slajdow :) ):
https://www.youtube.com/watch?v=cJDDsSj2vJA&t=829s

Reply:
Obejrzę jutro

Reply:
Chyba rozumiem już o co Tobie chodzi. Aktualnie  nie bedę w tym PR zmieniał tego